sound/soc/sof/intel/hda-stream.c

```diff
@@ -520,10 +504,45 @@ irqreturn_t hda_dsp_stream_threaded_handler(int irq, void *context)
 			/* Inform ALSA only in case not do that with IPC */
 			if (sof_hda->no_ipc_position)
 				snd_sof_pcm_period_elapsed(s->substream);
-
 		}
 	}
 
+	return active;
+}
+
+irqreturn_t hda_dsp_stream_threaded_handler(int irq, void *context)
+{
+	struct hdac_bus *bus = context;
+	u32 status;
+	u32 rirb_status;
+	int repeat = 0;
+	bool active = false;
+
+	status = snd_hdac_chip_readl(bus, INTSTS);
+
+	/*
+	 * repeat the loop 10 times to handle missed interrupts caused by
+	 * unsolicited responses from the codec
+	 */
+	do {
+		/* check streams */
+		active = hda_dsp_stream_check(bus, status);
+
+		/* check and clear RIRB interrupt */
+#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA)
+		if (status & AZX_INT_CTRL_EN) {
```

Comment:
Don't we want to re-read INTSTS in every loop iteration? Also this is now different from the previous version, where we checked against `stream_mask = GENMASK(sof_hda->stream_max - 1, 0) | AZX_INT_CTRL_EN`

Reply:
@ranj063 Agree with Guennadi, we should re-read INTSTS to have similar treatment for stream interrupts as well.

@lyakh That's ok, the old check was not correct. CTRL_EN (CIS) and stream bits (SIS) are different sources and should not be mixed in the logic.